controllers/awsmachine_controller.go

```diff
@@ -411,7 +410,7 @@ func (r *AWSMachineReconciler) reconcileNormal(_ context.Context, machineScope *
 	}
 
 	// Make sure Spec.ProviderID is always set.
-	machineScope.SetProviderID(fmt.Sprintf("aws:////%s", instance.ID))
+	machineScope.SetProviderID(instance.ID)
```

Comment:
Would it be better to update `getOrCreate` to also return the current AZ based on the returned AWS calls instead of trying to rely on the AWSMachine or Machine fields having FailureDomain set, since it's an optional field on both resources?

Reply:
Good idea. `getOrCreate` will end up converting `aws.instance` to `infrav1.instance` using `SDKToInstance`, maybe we can add AZ to `infrav1.Instance` and have conversion logic in `SDKToInstance` ? Is this an acceptable API change?

Reply:
Sounds good to me, I don't expect there to be any downstream consumers of that part of the API, so changes should not be an issue.